src/testset/tests/test_views.py

```diff
@@ -49,20 +53,36 @@ def test_success_passed(self):
 
         test = Test.objects.get(pk=PK)
         questions_count = test.questions_count()
+        user = User.objects.get(username=self.USERNAME)
         url = reverse('testset:next', kwargs={'pk': PK})
 
-        answers = [
-            {
-                'answer_3': "1",
-                'answer_4': "1"
-            },
-            {
-                'answer_3': "1",
-            },
-            {
-                'answer_2': "1",
-            },
-        ]
+        test_result = TestResult.objects.create(
+            user=user,
+            test=test
+        )
+
+        current_test_result = TestResult.objects.get(
+            id=test_result.id
+        )
+
+        answers = []
+
+        for question in test.questions.all():
+            answer = {}
+            for n, variant in enumerate(question.variants.all(), 1):
+                if variant.is_correct:
+                    answer[f'answer_{n}'] = "1"
+
+                TestResultDetail.objects.create(
+                    test_result=current_test_result,
+                    question=question,
+                    variant=variant,
+                    is_correct=True
+                )
```

Comment:
You don't need to create test results manually, then read it and check if they were displayed in HTML, because in this case test doesn't check business logic, but only if template was rendered correctly. This might be a interesting test on its own, for sure, but it would be related to UI. Let's just focus on business logic correctness in the first place.

Think of it as if you can _only_ use browser to interact with application and verify that it works correctly.

For this use `self.client.post` or `get` method to emulate user activities and `get` methods to verify results. Also, you might use read-only model methods (select queries) to verify those results that aren't available in UI. For example, `TestResult.correct_answers_count()` to get number of correct answers.